driver/configuration/provider_koanf.go

```diff
@@ -60,13 +59,18 @@ type (
 var _ Provider = new(KoanfProvider)
 
 func NewKoanfProvider(ctx context.Context, flags *pflag.FlagSet, l *logrusx.Logger, opts ...configx.OptionModifier) (kp *KoanfProvider, err error) {
+	cache, _ := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 100,
+		MaxCost:     1 << 22, // 4 MB
+		BufferItems: 64,
```

Comment:
This configuration is considered best practice for Ristretto:

```suggestion
		NumCounters:        maxItems * 10,
		MaxCost:            maxItems,
		BufferItems:        64,
		Metrics:            false,
		IgnoreInternalCost: true,
		Cost: func(value interface{}) int64 {
			return 1
		},
```

I would set maxItems to ~5000 which would cover 5000 different configs, more than enough